tests/builders/test_design_spaces.py

```diff
@@ -191,3 +193,99 @@ def test_exceptions(basic_cartesian_space, simple_mixture_space):
             name='invalid join space 2',
             description=''
         )
+
+
+def test_oversize_warnings(large_joint_design_space):
+    """Test that warnings are raised"""
+    # Test oversized formulation grid
+    with pytest.raises(UserWarning,
+                       match="Product design grid contains 1562500000 grid points. "
+                             "This may cause memory issues downstream."):
+        # Fail on error (so code stops running)
+        with warnings.catch_warnings():
+            warnings.simplefilter('error')
+            too_big_formulation_grid = {
+                'ing_F': np.linspace(0, 1),
+                'ing_G': np.linspace(0, 1),
+                'ing_H': np.linspace(0, 1),
+                'ing_I': np.linspace(0, 1),
+                'ing_J': np.linspace(0, 1),
+                'ing_K': np.linspace(0, 1)
+            }
+            enumerate_formulation_grid(
+                formulation_grid=too_big_formulation_grid,
+                balance_ingredient='ing_K',
+                name='too big mixture space',
+                description=''
+            )
+
+    # Test oversized enumerated space
```

Comment:
[Non-blocking] This test is completely independent of the "Test oversized formulation grid", so it would be better for it to be define in its own test.  That way, if "Test oversized formulation grid" failed but this test somehow didn't, you'd see that this test passed.  Also, although these tests are pretty light, breaking code like this into multiple tests provides more opportunity for parallel execution of tests.

Likewise with "Test oversized joined spaces"

Reply:
All separate now!